src/middleware/with-base-url.ts

```diff
@@ -0,0 +1,9 @@
+import { Middleware } from "retes";
+
+export const withBaseURL: Middleware = (handler) => async (request) => {
+  const { host, "x-forwarded-proto": protocol = "http" } = request.headers;
```

Comment:
I know its not a scope of refactor, but why http, not https?

Reply:
IIRC it's assuming http because of Docker routing.